src/tags/include.ts

```diff
@@ -40,4 +42,34 @@ export default class extends Tag {
     ctx.pop()
     ctx.restoreRegister(saved)
   }
+
+  public node (): StaticNode {
+    const values: Array<Value | ValueToken> = []
+    const blockScope: string[] = []
+
+    // TODO: jekyllInclude
+
+    for (const [k, v] of Object.entries(this.hash.hash)) {
+      blockScope.push(k)
+      if (isValueToken(v)) {
+        values.push(v)
+      }
+    }
+
+    if (isValueToken(this['file'])) {
+      values.push(this['file'])
+    }
+
+    if (isValueToken(this.withVar)) {
+      values.push(this.withVar)
+    }
+
+    return {
+      token: this.token,
+      values, // Values from this.hash and this.file
+      children: [],
+      blockScope, // Keys from this.hash and withVar
```

Comment:
same question for `blockScope` and `templateScope`, as they seem to be transient variables only needed for `analyze`, better make them computed instead of implemented in each tag.

But this issue is not big deal, I can help with it latter.

Reply:
As an example for where `blockScope` is necessary, I often use a non-standard `{% with %}` tag, inspired by [Django's template engine](https://docs.djangoproject.com/en/5.1/ref/templates/builtins/#with). It aliases and effectively caches long variable/property paths and keeps them in scope for the duration of its block.

For static analysis to work here, the tag must populate `blockScope` with those aliases, and in a way that informs the caller that these names go out of scope after `children` have been visited.

Reply:
Yeah, reasonable. As we'll need `blockScope` and `templateScope`, I don't think another `values` is not acceptable. Please treat my above comments as questions/ideas, I think what you're doing currently is also OK.